spec/importer_spec.rb

```diff
@@ -0,0 +1,178 @@
+    end
+
+    context 'when a block is given' do
+      before do
+        allow(output_file).to receive(:each).and_yield(status_record)
+      end
+
+      it 'must import status' do
+        expect(subject).to receive(:import_status)
+
+        subject.import(output_file) { |f| f }
+      end
+    end
+
+    context 'when no block is given' do
+      it 'must return an enumerator' do
+        expect(subject).to receive(:enum_for)
+
+        subject.import(output_file)
+      end
+    end
+  end
+
+  describe '.import_status' do
+    let(:status) { instance_double('Masscan::Status') }
+    let(:ip)     { "1.1.1.1" }
+
+    context 'when Masscan::Status status is :open' do
+      context 'when Masscan::Status protocol is :icmp' do
+        before do
+          allow(status).to receive(:status).and_return(:open)
+          allow(status).to receive(:protocol).and_return(:icmp)
+          allow(status).to receive(:ip).and_return(ip)
+        end
+
+        it 'must import ip address' do
+          expect(subject).to receive(:import_ip_address).with(ip)
+
+          subject.import_status(status)
+        end
+      end
+
+      context 'when Masscan::Status protocol is not :icmp' do
+        before do
+          allow(status).to receive(:status).and_return(:open)
+          allow(status).to receive(:protocol).and_return(:not_icmp)
+        end
+
+        it 'must import open port status' do
+          expect(subject).to receive(:import_open_port_status).with(status)
+
+          subject.import_status(status)
+        end
+      end
+    end
+  end
+
+  describe '.import_open_port_status' do
+    let(:status)    { instance_double('Masscan::Status', ip: '1.1.1.1', port: 80, protocol: :tcp, timestamp: Time.now) }
+    let(:ip_addr)   { instance_double('Ronin::DB::IpAddress') }
+    let(:port)      { instance_double('Ronin::DB::Port') }
+    let(:open_port) { instance_double('Ronin::DB::OpenPort') }
+
+    before do
+      Ronin::DB.connect
+      allow(subject).to receive(:import_ip_address).and_return(ip_addr)
+      allow(subject).to receive(:import_port).and_return(port)
+      allow(open_port).to receive(:update)
+      allow(Ronin::DB::OpenPort).to receive(:find_or_create_by).and_return(open_port)
+    end
+
+    context 'when block is given' do
+      it 'must yield open and return port' do
+        expect { |b|
+          subject.import_open_port_status(status, &b)
+        }.to yield_with_args(open_port)
+      end
+    end
+
+    it 'must return open port' do
+      expect(subject.import_open_port_status(status)).to eq(open_port)
+    end
+  end
+
+  describe '.import_ip_address' do
+    let(:ipv4_address)        { IPAddr.new('192.168.1.1') }
+    let(:ipv6_address)        { IPAddr.new('2001:0db8:85a3:0000:0000:8a2e:0370:7334') }
+    let(:imported_ip_address) { instance_double("Ronin::DB::IPAddress") }
+
+    before do
+      Ronin::DB.connect
+      allow(Ronin::DB::IPAddress).to receive(:find_or_create_by).and_return(imported_ip_address)
```

Comment:
This seems like too aggressive of stubbing. All we need to test is that an `Ronin::DB::IPAddress` is returned with the same IP address that we wish to import.